File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
static bool compare(pair<int,int>&p1,pair<int,int>&p2)
{
    return p1.second>p2.second;
}
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int>ans;
        unordered_map<int,int>m1;
        for(int x:nums)
        {
            m1[x]++;
        }
        vector<pair<int,int>>v;
        for(auto x:m1)
        {
            v.push_back({x.first,x.second});
        }
        sort(v.begin(),v.end(),compare);
        for(int i=0;i<k;i++)
        {
            ans.push_back(v[i].first);
        }
        return ans;
    }
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp=Counter(nums)
        res=[]
        ret=[]
        for x in mp.keys():
            res.append((mp[x],x))
        res.sort(reverse=True)
        for i in range(k):
            ret.append(res[i][1])
        return ret

        # import heapq
        # mp=Counter(nums)
        # heap=[]
        # ret=[]
        # for key,val in mp.items():
        #     heapq.heappush(heap,(val,key))
        #     if len(heap)>k:
        #         heapq.heappop(heap)
        # for i in heap:
        #     ret.append(i[1])
        # return ret

            
        


